### PR TITLE
- Handle logic with no internet connection

### PR DESCRIPTION
**Pull request by VolkovSwift**
Add Internet reachability
Fix UI

### CODE REVIEW

File: WishlistApp/Controllers/UsersViewController.swift.orig [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+//  UsersViewController.swift
+//  WishlistApp
+//
+//  Created by user on 8/15/20.
+//  Copyright © 2020 Vlad Volkov. All rights reserved.
+//
+
+import UIKit
+
+protocol UsersViewDeledate: AnyObject {
```
Comment (denisappdevelopment) [inline WishlistApp/Controllers/UsersViewController.swift.orig:11]: Check all .org files please

File: WishlistApp/Managers/ReachabilityManager.swift [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+class ReachabilityManager: ReachabilityManagerDelegate {
+    let reachability: Reachability
+
+    init() {
+        reachability = try! Reachability()
+    }
+
+    func checkConnection(completionHandler: @escaping (Result<AuthResult, WLError>) -> Void) {
+        reachability.whenReachable = { _ in
+//            if reachability.connection == .wifi {
```
Comment (denisappdevelopment) [inline WishlistApp/Managers/ReachabilityManager.swift:25]: Clean it please

File: WishlistApp/Utilities/UIViewController + Ext.swift [hunk 3]
```diff
@@ -84,4 +84,53 @@ extension UIViewController {
...
+        newView.translatesAutoresizingMaskIntoConstraints = false
+        infoLabel.translatesAutoresizingMaskIntoConstraints = false
+        infoLabel.text = "No Internet Connection"
+        infoLabel.textColor = .white
+        if #available(iOS 11.0, *) {
+            let guide = self.view.safeAreaLayoutGuide
+            newView.trailingAnchor.constraint(equalTo: guide.trailingAnchor).isActive = true
+            newView.leadingAnchor.constraint(equalTo: guide.leadingAnchor).isActive = true
+            newView.topAnchor.constraint(equalTo: guide.topAnchor).isActive = true
+            newView.heightAnchor.constraint(equalToConstant: 50).isActive = true
```
Comment (denisappdevelopment) [inline WishlistApp/Utilities/UIViewController + Ext.swift:109]: Let's make it less high, probably 30. And let's make it slightly transparent

File: WishlistAppTests/PresentersTests/MyWishlistPresenterTests.swift [hunk 4]
```diff
@@ -38,27 +38,26 @@ class MyWishlistPresenterTests: XCTestCase {
...
-        XCTAssertTrue(view.cellImageWasUpdated)
-    }
-
-    func testFailureUpdateCell() {
-        let url = URL(string: "Baz")!
-        let gift = Gift(name: "Foo", description: "", giftLink: "", price: "", photoURL: url)
-        sut.setImage(gift: gift, cell: GiftCell())
-        XCTAssertFalse(view.cellImageWasUpdated)
-    }
+//    func testGetGiftsSuccess() {
```
Comment (denisappdevelopment) [inline WishlistAppTests/PresentersTests/MyWishlistPresenterTests.swift:41]: Don't forget to uncomment it later

**Review by denisappdevelopment — CHANGES_REQUESTED** [review]
Let's remove orig files

File: WishlistAppTests/PresentersTests/MyWishlistPresenterTests.swift (hunk 4 above)
Comment (VolkovSwift) [reply WishlistAppTests/PresentersTests/MyWishlistPresenterTests.swift:41]: Okay, I'll remove it in new unit-tests branch

**Review by denisappdevelopment — APPROVED** [review]

**Pull request merged**